cms/templates/js/unit-access-editor.underscore

```diff
@@ -0,0 +1,133 @@
+<%
+var groupAccess = xblockInfo.get('group_access');
+var selectablePartitions = [];
+var groupIdToName = {};
+
+contentGroups.forEach(function (partition) {
+    if (partition['scheme'] != 'cohort' && partition['scheme'] != 'enrollment_track') {
+        return;
+    }
+    if (partition['scheme'] == 'enrollment_track') {
+        var tracks = [];
+        var trackSelected = false;
+        partition['groups'].forEach(function (group) {
+            if (group['selected']) {
+                trackSelected = true;
+            }
+        });
+        // Only add enrollment tracks if there are multiple tracks to restrict from
+        if (partition['groups'].length < 2 && !trackSelected) {
+            return;
+        }
+    }
+    selectablePartitions.push(partition);
+});
+
+selectablePartitions.forEach(function (partition) {
+    var schemeId = partition['id'];
+    var groups = partition['groups'];
+    groups.forEach(function (group) {
+        var groupId = group['id'];
+        var groupName = group['name'];
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        groupIdToName[fullId] = groupName;
+    })
+});
+
+var deletedGroups = [];
+
+for (var schemeId in groupAccess) {
+    groupAccess[schemeId].forEach(function (groupId) {
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        if (!groupIdToName[fullId]) {
+            deletedGroups.push(fullId);
+        }
+    })
+}
+
+// Need to include deleted groups logic to not change is_active boolean in contentstore/utils.py get_user_partition_info
```

Comment:
We can look into how difficult this is to fix, but yes the behavior will show in both dialogs, what we're currently running into you can reproduce from the following steps:

1. Create only a single Content Group on the Settings -> Group Configurations page, ex. "Content Group A"
2. In a new tab navigate to the Course Outline and open the unit Configuration option
3. On the Settings Group Configurations page, delete the content group you created, there should be none remaining and the You have not created any content groups yet. message should appear.
4. On the Course Outline tab with unit Configuration open, save the Unit configuration after selected the content group you had previously created and then deleted, and click save.
5. Refresh the page of the Course Outline tab, with a breakpoint on line 431 of contentstore/utils.py you'll only see the enrollment_track partitions from get_all_partitions_for_course.

The only way the previously Deleted Group appears is if there is at least one other content group created.